plugins/scheduled_task/lib/object_task_engine/KObjectTaskModifyEntryEngine.php

```diff
@@ -0,0 +1,217 @@
+<?php
+
+/**
+ * @package plugins.scheduledTask
+ * @subpackage lib.objectTaskEngine
+ */
+class KObjectTaskModifyEntryEngine extends KObjectTaskEntryEngineBase
+{
+	/**
+	 * @param KalturaBaseEntry $object
+	 */
+	function processObject($object)
+	{
+		/** @var KalturaModifyEntryObjectTask $objectTask */
+		$objectTask = $this->getObjectTask();
+		if (is_null($objectTask))
+			return;
+
+		KBatchBase::impersonate($object->partnerId);
+		$client = $this->getClient();
+		$metadataPlugin = KalturaMetadataClientPlugin::get($client);
+		$entryId = $object->id;
+		$outputMetadataProfileId = $objectTask->outputMetadataProfileId;
+		$outputMetadataArr = $objectTask->outputMetadata;
+		$inputMetadataProfileId = $objectTask->inputMetadataProfileId;
+		$inputMetadataArr = $objectTask->inputMetadata;
+		
+		$metadataFilter = new KalturaMetadataFilter();
+		$metadataFilter->metadataObjectTypeEqual = KalturaMetadataObjectType::ENTRY;
+		$metadataFilter->objectIdEqual = $entryId;
+		
+		if($outputMetadataProfileId != 0 && !empty($outputMetadataArr))
+		{
+			$entryResultForMetadataUpdate = $client->baseEntry->get($entryId);
```

Comment:
You should surround all API requests with try/catch to avoid an uncaught exception, this way you'll always be able to unimpersonate.